iroh-base/src/key.rs

```diff
@@ -485,7 +450,7 @@ mod tests {
 
     #[test]
     fn test_display_from_str() {
-        let key = SecretKey::generate();
```

Comment:
"As a follow up, all usage of SecretKey::generate in tests should be changed to use a seeded rng, like chacha8, instead of thread_rng"

So what's wrong with using thread_rng in tests? Nondeterminism, or something else. I agree that nondeterminism in tests is not great, but we have not been religious about this in the past.

Reply:
I think it's worth taking steps towards determinism.

Reply:
it's only about nondeterminism why I want to do this, we already have a lot of randomness from the network, so reducing it from factors like this is usually very helpful

Reply:
I kind off like the `SecretKey::from_bytes(&[1u8; 32])` approach for deterministic keys in tests.  But I agree that if you use only one key in a test I'm not sure why bother with it?